### PR TITLE
Fix migration race to create items in SQL

### PR DESCRIPTION
**Pull request by millerdev**
Race description:
- HQ process saves a Couch doc
- migration loads batch of Couch docs including that doc
- HQ process saves SQL doc corresponding to the Couch doc
- migration attempts to save batch of SQL docs, crash on duplicate
- resolution: remove that doc from the SQL batch and try again

### Safety story

Only affects Couch to SQL migration management commands. Tests added for lookup tables migration.

### Automated test coverage

Yes.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/fixtures/tests/test_couchsqlmigration.py [hunk 1]
```diff
@@ -383,6 +383,57 @@ def test_migration_with_old_doc_format(self):
             [],
         )
 
+    def test_migration_race(self):
```
Comment (millerdev) [inline corehq/apps/fixtures/tests/test_couchsqlmigration.py:386]: I'm not thrilled that this test will be deleted when the lookup tables migration is completed, but the code its testing is part of the migration framework, which will live on after that. It seems like a lot of overhead to create test model classes to simulate an in-progress migration, but maybe it's worth it? Thoughts?

cc @orangejenny

**Comment by millerdev** [comment]
After developing an [elaborate fix](https://github.com/dimagi/commcare-hq/pull/32033/commits/ea3e01836a12e8bd179362fdceb170b1d2eb5412) for the issue, I [realized](https://github.com/dimagi/commcare-hq/pull/32031) that SQL (and Django) already has a much [simpler and more efficient](https://github.com/dimagi/commcare-hq/pull/32033/commits/9f61231c3f61c4c1eb60465c09224a83fcf65658) way to achieve the same thing.

File: corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py [hunk 2]
```diff
@@ -364,10 +364,9 @@ def update_log(action, doc_ids):
             couch_class.wrap(doc)._migration_sync_to_sql(obj, save=False)
         if creates or updates:
             with transaction.atomic(), disable_sync_to_couch(sql_class):
-                sql_class.objects.bulk_create(creates)
-                if updates:
-                    for obj in updates:
-                        obj.save()
+                sql_class.objects.bulk_create(creates, ignore_conflicts=True)
```
Comment (dannyroberts) [inline corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py:367]: Reading the docs (https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create) my interpretation is that this flag will keep what's already in the db and discard the insert for any duplicate row. Sounds like what you want, just spelling it out for my understanding. I'm not sure I understand the race condition at a detailed enough level, but it sounds like the logic is that if the row is already there, then the mechanism keeping it up to date is working, so you'd never want to replace it with the "new" version in the `bulk_create`.

**Review by dannyroberts — APPROVED** [review]

File: corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py:367]: I am sure this should not be an issue but flagging that `ignore_conflicts` with bulk create is only available after Postgres 9.5.

Comment (dannyroberts) [reply corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py:367]: That should be fine. We announced upgrading to 9.6 and dropped support for 9.5 years ago, so this shouldn't be an issue.

**Review by AmitPhulera — APPROVED** [review]

File: corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py (hunk 2 above)
Comment (millerdev) [reply corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py:367]: > this flag will keep what's already in the db and discard the insert for any duplicate row. Sounds like what you want
> ...
> if the row is already there, then the mechanism keeping it up to date is working, so you'd never want to replace it with the "new" version in the `bulk_create`.

Correct. Although `bulk_create` will never replace/overwrite, it raises an error which previously caused the migration to crash.

**Pull request merged**